Data/Scripts/001_Technical/005_PluginManager.rb

```diff
@@ -726,4 +726,115 @@ def self.runPlugins
     end
   end
   #-----------------------------------------------------------------------------
+  #  Get plugin dir from name based on meta entries
+  #-----------------------------------------------------------------------------
+  def self.findDirectory(name)
+    # go through the plugins folder
+    Dir.get("Plugins").each do |dir|
+      next if !Dir.safe?(dir)
+      next if !safeExists?(dir + "/meta.txt")
+      # read meta
+      meta = self.readMeta(dir, "meta.txt")
+      return dir if meta[:name] == name
+    end
+    # return nil if no plugin dir found
+    return nil
+  end
+  #-----------------------------------------------------------------------------
+end
+#===============================================================================
+#  Class extension to alias functions with content from modules
+#    allows plugins to structure methods encapsulate components they're
+#    overwriting in separate modules, and automatically apply those methods
+#    to a desired class, aliasing any potential overlap
+#  Example usage:
+#
+#     module BattleExtensions
+#       def pbPlayer; end
+#       def pbNewFunc; end
+#     end
+#
+#     class Battle
+#       include PluginManager::Aliaser
+#       alias_with_module(BattleExtensions)
+#     end
+#
+#     Battle.new now has methods `pbNewFunc`, `pbPlayer` and `pbPlayer_old`
+#===============================================================================
+module PluginManager
+  module Aliaser
```

Comment:
I don't want to put this into Essentials. All it does is obscure aliasing and hides aliased method names, not to mention requires the user to put their new methods in some other module rather than the class they want them to be in (which is confusing).

Maybe you (Luka) find it a minor saving on keys typed, but I don't think it's worth having in Essentials. You can keep it in your own utilities plugin if you need it.

As a minor note, having a default extension of 'old' makes it possible to forget to write in an extension unique to the scripts you're writing. If two sets of scripts happen to alias the same method and neither adds their own extension (or both give the same extension), I think something's going to go missing because you just `next` if `method_name_old` is already defined. And even assuming both sets of scripts provide their own extensions, will `aliased_method` work properly in each version of the aliased method to call the version before it?

Reply:
Sure, that's fine. I'll remove the aliaser with the next commit.